feat(python): add support for defined names (named ranges)

AmdyTran opened the pull request:
## Summary
This PR adds support for reading Excel defined names (named ranges), similar to how OpenPyXL provides this functionality. This feature leverages the existing defined_names() method in the underlying calamine library.

## Changes
  - Added defined_names() method to ExcelReader that returns a list of tuples containing
  (name, formula) pairs
  - Implemented Rust bindings in src/types/excelreader/mod.rs
  - Added Python bindings and exposed the method through the Python API
  - Updated type stubs (.pyi file)

## Other
- Added tests
- Formula strings match Excel's format (e.g., "sheet1!$A$5:$D$7",  "SUM(sheet1!$K$5:$K$6)")
- Bumped version to 0.16

lukapeschke reviewed (changes requested):
Thanks a lot for you contribution! Could you please introduce a `DefinedName` type with properties ? We try to keep our APIs as explicit as possible 🙂 

For now, this could be something simple like 
```rust
struct DefinedName {
    pub name: String,
    pub formula: String,
}
```

AmdyTran commented:
> Thanks a lot for you contribution! Could you please introduce a `DefinedName` type with properties ? We try to keep our APIs as explicit as possible 🙂
> 
> For now, this could be something simple like
> 
> ```rust
> struct DefinedName {
>     pub name: String,
>     pub formula: String,
> }
> ```

Thanks, I updated it and I split the typing up etc, please have a look again.

lukapeschke commented on src/types/excelreader/mod.rs:
```diff
@@ -99,7 +111,17 @@ impl ExcelSheets {
     }
 }
-
+#[derive(Debug)]
+#[cfg_attr(feature = "python", pyclass(name = "DefinedName"))]
+pub struct DefinedName {
+    pub name: String,
+    pub formula: String,
+}
+impl From<(String, String)> for DefinedName {
+    fn from((name, formula): (String, String)) -> Self {
+        DefinedName { name, formula }
+    }
+}
```
```suggestion

#[derive(Debug)]
#[cfg_attr(feature = "python", pyclass(name = "DefinedName"))]
pub struct DefinedName {
    pub name: String,
    pub formula: String,
}

```

lukapeschke commented on src/types/excelreader/mod.rs:
```diff
@@ -58,6 +58,18 @@ impl ExcelSheets {
+        }
+        .to_vec()
+        .into_iter()
+        .map(DefinedName::from)
```
```suggestion
        .map(|(name, formula)| DefinedName { name, formula })
```

lukapeschke commented on python/tests/test_defined_names.py:
```diff
@@ -0,0 +1,24 @@
+
+    # Convert to dict for easier checking
+    names_dict = {dn.name: dn.formula for dn in defined_names}
+
+    assert "DefinedRange" in names_dict
+    assert names_dict["DefinedRange"] == "sheet1!$A$5:$D$7"
+
+    assert "NamedConstant" in names_dict
+    assert names_dict["NamedConstant"] == "3.4"
+
+    assert "AddingValues" in names_dict
+    assert names_dict["AddingValues"] == "SUM(sheet1!$K$5:$K$6)"
```
Could you please directly compare to the expected dict ? This would ensure that we don't have any extra key in the dict
